### PR TITLE
[TRAFODION-3320] support multiple characters field and record separator when for data loading and extracting

### PR DESCRIPTION
**Pull request by SuJinpei**
**changes:**

1. add two options: 'mcfs' and 'mcrs' to specify the multi-character field separator and record separator.
2. add two functions: 'replace_escapes' and 'hextobin' to handle escape sequences in the separator.
3. currentFiledPos -> currentFieldPos 

**separator check algorithm:**
  check last few characters in temp field buffer to see if we found a field separator and record separator.

### CODE REVIEW

File: core/conn/odb/src/odb.c [hunk 1]
```diff
@@ -1537,7 +1540,9 @@ int main(int ac, char *av[])
             fprintf(stderr, "\tChild EID (.child): %d\n", etab[i].child);
             fprintf(stderr, "\tTBE flag (.tbe): %d\n", etab[i].tbe);
             fprintf(stderr, "\tField Separator (.fs): %d (decimal value)\n", etab[i].fs);
+            fprintf(stderr, "\tMuti Charaters Field Separator (.mcfs): %s\n", etab[i].mcfs);
```
Comment (DaveBirdsall) [inline core/conn/odb/src/odb.c]: Typo, I assume you meant "Multi Characters ..." here.

File: core/conn/odb/src/odb.c [hunk 2]
```diff
@@ -1537,7 +1540,9 @@ int main(int ac, char *av[])
...
             fprintf(stderr, "\tRecord Separator (.rs): %d (decimal value)\n", etab[i].rs);
+            fprintf(stderr, "\tMuti Charaters Record Separator (.mcrs): %s\n", etab[i].mcrs);
```
Comment (DaveBirdsall) [inline core/conn/odb/src/odb.c]: Similar typo here

File: core/conn/odb/src/odb.c [hunk 3]
```diff
@@ -14947,26 +15052,42 @@ static void usagexit()
         "      [:xmltag=[+]element][:xmlord][:xmldump]\n"
 #endif
         "      Defaults/notes:\n"
-        "      * src file: local file or {hdfs,mapr}[@host,port[,huser]].<HDFS_PATH>\n"  
+        "      * src file: local file or {hdfs,mapr}[@host,port[,huser]].<HDFS_PATH>\n"
         "      * fs: default ','. Also <ASCII_dec> 0<ASCII_OCT> X<ASCII_HEX>\n"
-        "      * rs: default '\\n'. Also <ASCII_dec> 0<ASCII_OCT> X<ASCII_HEX>\n" 
-        "      * ec: default '\\'. Also <ASCII_dec> 0<ASCII_OCT> X<ASCII_HEX>\n" 
-        "      * pc: no default. Also <ASCII_dec> 0<ASCII_OCT> X<ASCII_HEX>\n" 
+        "      * mcfs: default use fs as feild seperator. If mcfs was set then use mcfs as filed\n"
```
Comment (DaveBirdsall) [inline core/conn/odb/src/odb.c]: Typo, "feild" should be "field"

File: core/conn/odb/src/odb.c [hunk 4]
```diff
@@ -14947,26 +15052,42 @@ static void usagexit()
...
+       "      [:ec=escape_char][:soe][:ns=nullstring][:es=emptystring][:rows=#rowset][:nomark]\n"
+       "      [:binary][:bpc=#][:bpwc=#][:max=#max_rec][:[r]trim[+]][:cast][:multi][:parallel=number]\n"
+       "      [:gzip[=lev]][:splitby=column][:uncommitted][:iobuff=#size][:hblock=#size][:ucs2toutf8]\n"
        "      [:pre={@sqlfile|sqlcmd}[:mpre={@sqlfile|sqlcmd}[:post={@sqlfile|sqlcmd}]\n"
        "      [:tpar=#tables][:time][:cols=[-]columns]][:maxlen=#bytes][:xml]\n"
         "      Defaults/notes:\n"
         "      * tgt file: local file or {hdfs,mapr}.[@host,port[,huser]].<HDFS_PATH>\n"  
         "      * fs: default ','. Also <ASCII_dec> 0<ASCII_OCT> X<ASCII_HEX>\n"
-        "      * rs: default '\\n'. Also <ASCII_dec> 0<ASCII_OCT> X<ASCII_HEX>\n" 
+        "      * mcfs: default use fs as feild seperator. If mcfs was set then use mcfs as filed\n"
```
Comment (DaveBirdsall) [inline core/conn/odb/src/odb.c]: Typo, "feild"

Comment (SuJinpei) [reply core/conn/odb/src/odb.c]: thank Dave.

**Pull request merged**